front/elephant/src/app/components/share-note-dialog/share-note-dialog.component.ts

```diff
@@ -16,25 +16,28 @@ export class ShareNoteDialogComponent implements OnInit, OnDestroy {
   shareSubscription: Subscription;
 
   constructor(
+    @Inject(MAT_DIALOG_DATA) public data: any,
     private formBuilder: FormBuilder,
     private noteService: NoteService,
     private snackBar: SnackbarService,
-    public dialogRef: MatDialogRef<ShareNoteDialogComponent>,
-  ) { }
+    public dialogRef: MatDialogRef<ShareNoteDialogComponent>
+  ) {
+  }
 
   ngOnInit(): void {
 
     this.shareForm = this.formBuilder.group({
       email: ['', [Validators.required, Validators.email]],
-      mode: ['read', Validators.required]
+      permissionType: ['Viewer', Validators.required]
     });
   }
+
   ngOnDestroy(): void {
     this.shareSubscription?.unsubscribe();
   }
 
   onSubmit(): void {
-    this.shareSubscription = this.noteService.shareNote(this.shareForm.value).subscribe(res => {
+    this.shareSubscription = this.noteService.shareNote(this.data.noteId, this.shareForm.value).subscribe(res => {
```

Comment:
Agregar un chequeo con HttpStatus.NOT_FOUND, "User Not Found" para mostrar un mensaje de error "No existe un usuario con ese email"
Agregar un chequeo con HttpStatus.FORBIDEN, "User already has a permission over this note" para mostrar un mensaje de error "El usuario ya tiene un permiso sobre esta nota"